lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -1349,6 +1317,12 @@ bool GenericSignatureBuilder::addSameTypeRequirementBetweenArchetypes(
     T1->ConcreteTypeSource = T2->ConcreteTypeSource;
   }
 
+  // Decide which potential archetype is to be considered the representative.
```

Comment:
I'm curious why this needs to move, the code above seems like it shouldn't care?